Library/Homebrew/blacklist.rb

```diff
@@ -83,5 +83,8 @@ def blacklisted?(name)
     A binary installer is available:
       https://www.haskell.org/platform/mac.html
     EOS
+  when "mysqldump-secure" then <<-EOS.undent
+    The creator of mysqldump-secure tried to game our popularity metrics.
```

Comment:
@mikemcquaid Let me put it this way:
I know I've done something wrong and I'm sorry for the trouble caused.
This is also the reason why I'm here trying to give some suggestions with my initial thinking in mind. 
As I have already said, the blacklisting is a fair thing.

From what I have read so far is that this happens very rarely.
If those occurrences are mostly self-submissions then I would favour your initial idea to check against self-submissions.